### PR TITLE
CLOUDP-149154: Added import for Global deployment

### PR DESCRIPTION
**Pull request by Sugar-pack**
## Proposed changes

Added an ability to export private endpoints of Atlas global cluster for the `kubernetes config generate` command

**NOTE**: `mongodb-atlas-kubernetes` dependency is not set to the release version, because the feature is not released yet. A proper version (1.6.0) will be set as soon as it will be released. 
It is also necessary to wait for the new version of the [go-client-mongodb-atlas](https://github.com/mongodb/go-client-mongodb-atlas) because it will contain fixes for managed namespaces and advanced deployments.  https://github.com/mongodb/go-client-mongodb-atlas/pull/337

_Jira ticket:_ CLOUDP-149154

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: go.mod
Comment (andreaangiolillo) [inline go.mod:39]: We generally make a new go client release when we need it instead of pinging a specific not-released commit. I will release a 0.20.2 today so that you can use that release.

File: go.mod
Comment (andreaangiolillo) [inline go.mod:28]: [q] when will v1.5.3 be released?

File: internal/kubernetes/operator/deployment/deployment.go
Comment (andreaangiolillo) [inline internal/kubernetes/operator/deployment/deployment.go:78]:
```suggestion
	customZoneMapping, managedNamespaces, err := buildGlobalDeployment(deployment.ReplicationSpecs, deploymentStore, projectID, clusterID)
```

File: internal/kubernetes/operator/deployment/deployment.go
Comment (andreaangiolillo) [inline internal/kubernetes/operator/deployment/deployment.go:95]:
```suggestion
		ManagedNamespaces:        managedNamespaces,
```

File: internal/kubernetes/operator/deployment/deployment.go
Comment (andreaangiolillo) [inline internal/kubernetes/operator/deployment/deployment.go:179]:
```suggestion
	if globalCluster.ManagedNamespaces == nil {
	        return customZoneMapping,nil, nil
	}
	
	for _, ns := range globalCluster.ManagedNamespaces {
	     managedNamespace = append(managedNamespace, atlasV1.ManagedNamespace{
	     Db:                     ns.Db,
	     Collection:             ns.Collection,
	     CustomShardKey:         ns.CustomShardKey,
	     NumInitialChunks:       ns.NumInitialChunks,
	     PresplitHashedZones:    ns.PresplitHashedZones,
	     sCustomShardKeyHashed: ns.IsCustomShardKeyHashed,
	     sShardKeyUnique:       ns.IsShardKeyUnique,
	}}

	return customZoneMapping, managedNamespace, nil
```

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (andreaangiolillo) [inline internal/kubernetes/operator/deployment/deployment_test.go:45]: no need to define this mock. We use `mockgen` and you already added the correct mock annotation in internal/store/global_deployment.go. So if you run `make gen-mocks`, the mock is autogenerated and can be used in your test

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (andreaangiolillo) [inline internal/kubernetes/operator/deployment/deployment_test.go:43]: [q] Can you update the e2e test as well?

File: go.mod
Comment (Sugar-pack) [reply go.mod:39]: Thank you

File: go.mod
Comment (Sugar-pack) [reply go.mod:28]: The release is planned for early/mid-February. But I will ask the team if we can release the release sooner.

File: go.mod
Comment (andreaangiolillo) [reply go.mod:39]: I will release it this afternoon. Waiting for the go-ahead from terraform

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: Some errors with project deletion. 	Error: DELETE https://cloud-dev.mongodb.com/api/public/v1.0/groups/1234: 500 (request "UNEXPECTED_ERROR") Unexpected error.
[2023/01/09 10:12:12.802]     projects_test.go:121: got="Error: DELETE https://cloud-dev.mongodb.com/api/public/v1.0/groups/1234: 500 (request \"UNEXPECTED_ERROR\") Unexpected error.\n"

File: go.mod
Comment (andreaangiolillo) [reply go.mod:39]: v0.21.0 has been released

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (andreaangiolillo) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: No, the project deletion error are not related to the library upgrade but to some endpoint in cloud-dev that are not working as expected. You should still be able to update the e2e test [kubernetes_config_generate_test.go](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/e2e/atlas/kubernetes_config_generate_test.go)

File: internal/store/global_deployment.go
Comment (andreaangiolillo) [inline internal/store/global_deployment.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

**Review by andreaangiolillo — COMMENTED** [review]
A few comments are not addressed: mock and e2e test

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: Global cluster is a feature for advanced and legacy clusters. And as I see, we do not write e2e tests for each feature.

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:45]: Let's merge it as is. But we will create a ticket to rewrite our tests with autogenerated mocks. Is it ok?

Comment (andreaangiolillo) [reply internal/kubernetes/operator/deployment/deployment_test.go:45]: sure, can you share a link to the ticket and ETA? just want to make sure that this will be addressed

File: go.mod
Comment (andreaangiolillo) [reply go.mod:28]: Regarding this:

> mongodb-atlas-kubernetes dependency is not set to the release version, because the feature is not released yet. A proper version (1.6.0) will be set as soon as it will be released.

I think we may want to wait for 1.6.0 to be released before merging this PR. AtlasCLI must be always ready to be released at any time and we cannot release atlascli with this feature without the kub 1.6.0.

File: go.mod
Comment (andreaangiolillo) [inline go.mod:39]: we have recently upgraded atlas go client in atlascli. Can you update your PR with the latest changes from the repo?

File: internal/cli/atlas/security/ldap/save.go
Comment (andreaangiolillo) [inline internal/cli/atlas/security/ldap/save.go:113]: same for this. We have upgraded the go client in master and added other fixes as well.

File: internal/store/global_deployment.go
Comment (andreaangiolillo) [inline internal/store/global_deployment.go:30]: Let's keep the same name as the go client
```suggestion
func (s *Store) GlobalCluster(projectID, instanceName string) (*atlas.GlobalCluster, error) {
```

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (andreaangiolillo) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: at this line [kubernetes_config_generate_test](https://github.com/mongodb/mongodb-atlas-cli/blob/d72fcee764cdd38f3831ae10b93d13ffb9d04157/test/e2e/atlas/kubernetes_config_generate_test.go#L70), we are setting up a cluster for the e2e test so you should be able to test that the `kub generate command` has created the correct yaml for the cluster right?

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:45]: https://jira.mongodb.org/browse/CLOUDP-152961 
I will do this task as soon as I finish the current one. So I will create a PR with a test update on Friday or Monday.

Comment (andreaangiolillo) [reply internal/kubernetes/operator/deployment/deployment_test.go:45]: Thanks!

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: Yeah. But now our tests compare only the names of the resources. I agree that we should add e2e for each feature, but let's do it in another PR. Do you mind?

File: go.mod
Comment (Sugar-pack) [reply go.mod]: Done

File: internal/cli/atlas/security/ldap/save.go
Comment (Sugar-pack) [reply internal/cli/atlas/security/ldap/save.go]: Done

File: internal/kubernetes/operator/deployment/deployment_test.go
Comment (andreaangiolillo) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: SGTM. Could you create a ticket if you are not planning to address this as part of [CLOUDP-152961](https://jira.mongodb.org/browse/CLOUDP-152961)? Thanks!

Comment (Sugar-pack) [reply internal/kubernetes/operator/deployment/deployment_test.go:43]: Done. [CLOUDP-153348](https://jira.mongodb.org/browse/CLOUDP-153348)

File: go.sum
Comment (gssbzn) [inline go.sum:302]: can you please run `go mod tidy`

Comment (Sugar-pack) [reply go.sum]: Done

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**
